### PR TITLE
Implement Xcode misplaced finder

### PR DESCRIPTION
**Pull request by giginet**
I added new Xcode tool to find misplaced constraints on Storyboard/XIB.

Look giginet/dokumi-test#2 to check behavior of this feature

screen shot 2016-03-29 at 17 27 27

### CODE REVIEW

File: lib/dokumi/support/xml.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+module Dokumi
+  module Support
+    module XML
+      def self.read_xml_tag(text)
+        h = {}
+        s = StringScanner.new(text)
```
Comment (slightair) [inline lib/dokumi/support/xml.rb]: You can use Nokogiri.

Comment (giginet) [inline lib/dokumi/support/xml.rb]: This implementation is already exist and I just moved from the another module.
I think we should refactor it in other PR.

Comment (vincentisambart) [inline lib/dokumi/support/xml.rb]: As slightair was saying, you should use Nokogiri.
The reason `UnchangedStoryboardFinder` was parsing the XML without using Nokogiri was because it was working on diffs, not on whole files. `MisplacedConstraintFinder` should work on whole files.

File: lib/dokumi/tool/xcode/misplaced_constraint_finder.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
+module Dokumi
+  module Tool
+    class Xcode
+      MisplacedStatus = Struct.new('MisplacedStatus', :file_path, :line_no)
```
Comment (vincentisambart) [inline lib/dokumi/tool/xcode/misplaced_constraint_finder.rb]: Should probably be in `MisplacedConstraintFinder`, not `Xcode`.

Comment (vincentisambart) [inline lib/dokumi/tool/xcode/misplaced_constraint_finder.rb]: Most of the Dokumi codebase uses normal hashes, not structs for this type of code.

File: lib/dokumi/tool/xcode/misplaced_constraint_finder.rb [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+      # Checks if a pull request contains XIB/Storyboard files which have misplaced constraints.
+      class MisplacedConstraintFinder
+        def self.find_issues(environment)
+          local_copy = environment.options[:local_copy]
+          raise "The local copy information is needed to find unchanged files." unless local_copy
+          diff = local_copy.diff_with_merge_base
+          misplaced_statuses = []
+          diff.rugged_diff.each_patch do |patch|
```
Comment (vincentisambart) [inline lib/dokumi/tool/xcode/misplaced_constraint_finder.rb:10]: `MisplacedConstraintFinder` probably doesn't need to work on diffs. Just add issues, and Dokumi will automatically filter out the ones that are not related to the current PR.
If `MisplacedConstraintFinder` takes too long, using the diff to make a list of files to parse might be a good idea, but parsing just the lines concerned just makes the code more complicated.

File: lib/dokumi/tool/xcode/misplaced_constraint_finder.rb [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+                  misplaced_statuses << misplaced_status
+                end
+              end
+            end
+          end
+          misplaced_statuses.each do |status|
+            environment.add_issue(
+              file_path: status.file_path,
+              line: diff.file_line_to_diff_line(status.file_path)[status.line_no],
+              type: :error,
```
Comment (vincentisambart) [inline lib/dokumi/tool/xcode/misplaced_constraint_finder.rb]: A warning would probably be more appropriate.
Also adding a tool name might be a good idea: `tool: :misplaced_constraint_finder,`

File: lib/dokumi/tool/xcode/misplaced_constraint_finder.rb [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+              description: "This constraints is misplaced.",
```
Comment (vincentisambart) [inline lib/dokumi/tool/xcode/misplaced_constraint_finder.rb]: `constraints` → plural
`is` → singular
s/constraints/constraint

**Comment by giginet** [comment]
@vincentisambart @slightair Thank you for reviewing.

I use Nokogiri to parse XML and refactored reviewed points.

Please check it out.

**Comment by slightair** [comment]
LGTM

**Comment by giginet** [comment]
@slightair Thank you for reviewing.
Can I merge it? @vincentisambart

**Comment by vincentisambart** [comment]
LGTM
